modules/mobile-agents/android/client/plugins/IDPProxy/src/org/wso2/mdm/agent/proxy/RefreshTokenHandler.java

```diff
@@ -0,0 +1,149 @@
+			APIUtilities apiUtilities = new APIUtilities();
+			apiUtilities.setEndPoint(IdentityProxy.getInstance().getAccessTokenURL());
+			apiUtilities.setHttpMethod("POST");
+			apiUtilities.setRequestParamsMap(request_params);
+
+			Map<String, String> headers = new HashMap<String, String>();
+			Log.e("proxy", IdentityProxy.clientID + ":" + IdentityProxy.clientSecret);
+			String authorizationString =
+					"Basic " +
+					new String(
+							Base64.encodeBase64((IdentityProxy.clientID +
+							                     ":" + IdentityProxy.clientSecret).getBytes())
+					);
+			headers.put("Authorization", authorizationString);
+			headers.put("Content-Type", "application/x-www-form-urlencoded");
+
+			Map<String, String> response_params =
+					ServerApiAccess.postDataAPI(apiUtilities, headers);
+
+			response = response_params.get("response");
+			responseCode = response_params.get("status");
+			Log.d(TAG, response);
+			return response;
+		}
+
+		@SuppressLint("SimpleDateFormat")
+		@Override
+		protected void onPostExecute(String result) {
+
+			String refreshToken = null;
+			String accessToken = null;
+			int timeToExpireSecond = 3000;
+			IdentityProxy identityProxy = IdentityProxy.getInstance();
+			try {
+				JSONObject response = new JSONObject(result);
+				Log.e("refresh Token Post", result.toString());
+
+				if (responseCode != null && responseCode.equals("200")) {
+					refreshToken = response.getString("refresh_token");
+					accessToken = response.getString("access_token");
+					timeToExpireSecond = Integer.parseInt(response.getString("expires_in"));
+
+					token.setRefreshToken(refreshToken);
+					token.setAccessToken(accessToken);
+
+					SharedPreferences mainPref =
+							IdentityProxy.getInstance()
+							             .getContext()
+							             .getSharedPreferences("com.mdm",
+							                                   Context.MODE_PRIVATE);
+					Editor editor = mainPref.edit();
+					editor.putString("refresh_token", refreshToken);
+					editor.putString("access_token", accessToken);
+
+					DateFormat dateFormat = new SimpleDateFormat("MM/dd/yyyy HH:mm:ss");
+					Date date = new Date();
+					long expiresIN = date.getTime() + (timeToExpireSecond * 1000);
+					Date expireDate = new Date(expiresIN);
+					String strDate = dateFormat.format(expireDate);
+					token.setDate(strDate);
+					editor.putString("date", strDate);
+					editor.commit();
+
+					identityProxy.receiveNewAccessToken(responseCode, "success", token);
+
+				} else if (responseCode != null) {
+					JSONObject mainObject = new JSONObject(result);
+					String error = mainObject.getString("error");
+					String errorDescription = mainObject.getString("error_description");
+					Log.d(TAG, error);
+					Log.d(TAG, errorDescription);
+					identityProxy.receiveNewAccessToken(responseCode, errorDescription, token);
+				}
+			} catch (JSONException e) {
+				identityProxy.receiveNewAccessToken(responseCode, "", token);
+				e.printStackTrace();
```

Comment:
Need to get rid of printing stacktraces. Instead use proper logging and exception handling within the catch block.